### ISSUE TITLE
A gettext call containing placeholders was found ...

### ISSUE THREAD

**Issue by eugelogic** [WordPress-Coding-Standards/WordPress-Coding-Standards#1029]
Morning all, could any of you please shed some light on the following?
I asked for help on the #core-coding-standard Slack channel but it looked like I had to open an issue here on GitHub.
I’m running my code against the WPCS and bumped into the following error:
`A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders. (WordPress.WP.I18n.MissingTranslatorsComment)`
I then added the following to my code to remove the error:
`/* translators: %s: search term */`
when I run WPCS the error is still there, am I missing something obvious here?
FYI I’m using Underscores and there are similar files (in the templates-parts folder) with the same comments that do not throw that error.
the code I need to comment upon is:
`<h2 class="page-title"><?php printf( esc_html__( 'Search Results for: %s', 'hlcchl' ), '<span>' . get_search_query() . '</span>' ); ?></h2>`
the whole file is here [https://github.com/eugelogic/HLCCHL-with-Bootstrap-Sass/blob/master/themes/hlcchl/search.php#L22](https://github.com/eugelogic/HLCCHL-with-Bootstrap-Sass/blob/master/themes/hlcchl/search.php#L22)
Many thanks in advance,
Eugene

**Comment by jrfnl** [comment]
You've added the comment within the inline HTML and it will display on your page now. You need to add it in PHP, i.e. `<?php /* ... */ ?>`

**Comment by eugelogic** [comment]
DOH &#128542;
schoolboy error - thanks for getting back to me @jrfnl

**Comment by GaryJones** [comment]
@eugelogic As Juliette says, try:

```php
<header class="page-header">
	<?php /* translators: %s: search term */ ?>
	<h2 class="page-title"><?php printf( esc_html__( 'Search Results for: %s', 'hlcchl' ), '<span>' . get_search_query() . '</span>' ); ?></h2>
</header><!-- .page-header -->
```

or

```php
<header class="page-header">
	<h2 class="page-title"><?php
	/* translators: %s: search term */
	printf( esc_html__( 'Search Results for: %s', 'hlcchl' ), '<span>' . get_search_query() . '</span>' );
	?></h2>
</header><!-- .page-header -->
```

**Comment by eugelogic** [comment]
Thanks @GaryJones, I'll settle for the first option

**Comment by GaryJones** [comment]
My own preference would be to separate out the escape + translate + string + get_search_query from the HTML as much as possible:

```php
<?php
	/* translators: %s: search term */
	$page_title = sprintf(
		__( 'Search Results for: %s', 'hlcchl' ),
		'<span>' . get_search_query() . '</span>'
	);
?>

<header class="page-header">
	<h2 class="page-title"><?php echo esc_html( $page_title ); ?></h2>
</header>
```

The PHP can now be moved anywhere, and the markup that's left is considerably cleaner.

**Comment by eugelogic** [comment]
Thanks @GaryJones for the extra input, very informative.

**Comment by BaseerSatti1234** [comment]
for example use this to solve your issue 
	<?php
													// Translators: 1 is the opening <em> tag, 2 is the closing </em> tag.//
